src/objects/zcl_abapgit_object_type.clas.abap

```diff
@@ -162,13 +146,15 @@ CLASS zcl_abapgit_object_type IMPLEMENTATION.
 
   METHOD zif_abapgit_object~exists.
 
-    TRY.
-        read( ).
-        rv_bool = abap_true.
-      CATCH zcx_abapgit_not_found
-            zcx_abapgit_exception.
-        rv_bool = abap_false.
-    ENDTRY.
+    DATA: ls_tadir TYPE tadir.
+
+    SELECT SINGLE * FROM tadir INTO ls_tadir
+      WHERE pgmid    = 'R3TR'
+        AND object   = ms_item-obj_type
+        AND obj_name = ms_item-obj_name.
+    IF ls_tadir IS NOT INITIAL.
```

Comment:
Minor, but probably better to read a single column (e.g. `object`) instead of `SELECT *`

Reply:
that select will always work since only objects in tadir are serialized. `exists` needs to check the actual object definition (like DD* table).

Reply:
Can't find DD* table for object definition. However, text definitions are stored in DDTYPET. Selecting text definition seems to be not a good idea for existence check. The next best option would be to query PROGDIR table. For a table type with name Y001, program name exists with **%_C**Y001

Reply:
Progdir is good

Reply:
Changes made